lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -167,18 +188,22 @@ bool ArrayAllocation::isInitializationWithKnownElements() {
 ///
 /// We have to prove that the array value is not changed in between the
 /// creation and the method call to get_element.
-bool ArrayAllocation::findValueReplacements() {
+bool ArrayAllocation::findReplacements() {
   if (!isInitializationWithKnownElements())
     return false;
 
   // The array value was stored or has escaped.
   if (!analyseArrayValueUses())
     return false;
 
-  // No count users.
-  if (GetElementCalls.empty())
+  if (GetElementCalls.empty() && AppendContentsOfCalls.empty())
     return false;
 
+  for (auto *call : AppendContentsOfCalls) {
+    AppendContentsOfReplacementMap.push_back(
+        std::make_pair(call, ElementValueVector));
+  }
+
```

Comment:
You should add some upper bound on the number of elements, so that we don't explode code size if a very large array is appended.